update aten op overload to not use `from` to avoid compile errors

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89797
* #94611

Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.

eellison committed:
update from signature

[ghstack-poisoned]

eellison committed:
Update on "update from signature"




[ghstack-poisoned]

eellison committed:
Update on "update from signature"




[ghstack-poisoned]

tugsbayasgalan commented:
You need to run mobile script to regenerate upgrader bytecodes i think

tugsbayasgalan reviewed (changes requested):
Missing new upgrader bytecode

eellison committed:
Update on "update from signature"


Fix for https://github.com/pytorch/torchdynamo/issues/1795 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

cccclai commented on aten/src/ATen/native/ts_native_functions.yaml:
```diff
@@ -102,7 +102,7 @@ full_codegen:
   - pow.Tensor_Tensor
   - random
-  - random.from
+  - random.from_int
```
Trying to follow, are we deleting one operator (`random.from`) and add an operator (`random.from_int`) in this pr?

cccclai replied:
By browsing the previous pr, seems like we are trying to update `aten::random_.from`?

eellison replied:
changing `random_.from to random_.from_int` as noted in description

cccclai replied:
I think upgrader is used for one operator, like changing schema or semantic. If it's renamed, we are doing two operations here:

1. delete one operator
2. add a new operator

I wonder which case will be easier for you

option 1. keep the same op name `random_.from` and change the schema (or semantic)
option 2. add a new operator `random_.from_int` and keep the original operator `random_.from` as is.

@tugsbayasgalan maybe can confirm.

eellison replied:
Why would doing two changes be different than doing one change at a time in two separate PRs ?

eellison committed:
Update on "update from signature"


Fix for https://github.com/pytorch/torchdynamo/issues/1795 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison committed:
Update on "update from signature"


Fix for https://github.com/pytorch/torchdynamo/issues/1795 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

ezyang commented:
So, are you going to finish this?

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison commented:
cc @JackCaoG @Jiaw I have an XLA failure here because I'm updating the name of an operator so it can be printed in python without a syntax error. Would you advise me on how to move forward with landing ?

JackCaoG commented:
@wonjoolee95 Can you follow up?

wonjoolee95 commented:
@eellison, I'll work on a PR from XLA's side to fix this and comment here when it's ready. Should be ready very soon, just need to rebuild my local pytorch.

wonjoolee95 commented:
@eellison, the XLA's PR is created. Could you update the XLA commit pin in the PR to `a121c7d3353f1c313ddc0fc97cc41162a3dd28e4`? 

Once CIs are green on both this PR and XLA's PR, we can coordinate a merge to keep both heads green -- once this PR's CI is green, rebase this PR with master and force merge (with `merge -f`) flag with the XLA pin as is. Note that if you don't have permission to force merge, we'll need someone else's help who is authorized. Then I'll follow-up with merging the XLA's side PR.

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison commented:
@wonjoolee95 this is passing (with the exception of one flakey test). let me know when to merge

wonjoolee95 commented:
@eellison, XLA's side is ready to merge. We can rebase this PR with master one last time and force merge. Then I'll follow-up with merging XLA's PR.

eellison commented:
@wonjoolee95 is it necessary to rebase ? The tests take a while to run. Or are you suggesting to rebase and not wait for tests

wonjoolee95 commented:
Yes to both -- generally, we rebase one last time before we merge these PRs that touch XLA commit pins. Because if we don't, the  merge will often to fail due to merge conflict in the XLA pin (the pin might have changed in the PyTorch master vs. this PR's old pin). 

After the rebase, we can force merge (without waiting for tests) since we know this PR's CIs were all green.

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison commented:
@wonjoolee95  rebased - let me know when you're ready for me to merge

wonjoolee95 commented:
> @wonjoolee95 rebased - let me know when you're ready for me to merge

Ready!

desertfire commented:
@eellison , thanks a lot for pushing this through!

jeanschmidt commented:
@pytorchbot revert -m "breaking internal builds - more details on https://fburl.com/sandcastle/bz8mgkil" -c nosignal

JackCaoG commented:
@wonjoolee95 I revert the XLA side PR, can you coordinate the reland on our end again? Thanks.

eellison commented:
@JackCaoG sorry about that!

eellison committed:
Update on "update aten op overload to not use `from` to avoid compile errors"


Fix for https://github.com/pytorch/pytorch/issues/93591 by changing `random_.from` to `random_.from_int`.

The previous signature would fail when printed in an fx graph, because `from` is a reserved python keyword. This change affects serialization but I have added an adapter.
 

[ghstack-poisoned]

eellison commented:
I ran into an FC issue with `random_.from_int` not being defined. Unfortunately, `random_.from` and `random_.from_int` (with same signature) cannot exist concurrently because that breaks all sorts of invariants with the build. 

Moving forward I plan on adding a 
`random_.from -> rand_.from_int` upgrader, waiting a couple weeks for the FC to pass, then going from `rand_ -> random_.from_int`. This is a bit circuitous but I dont know of a better solution.

wonjoolee95 commented:
Please tag me when this is ready to merge, thanks!
